src/app/containers/PageHandlers/withVariant/index.jsx

```diff
@@ -1,50 +1,41 @@
-import PropTypes from 'prop-types';
-import pathOr from 'ramda/src/pathOr';
 import React from 'react';
 import { useParams, useLocation, Redirect } from 'react-router-dom';
-import getVariantRedirectUrl from './getVariantRedirectUrl';
+import { variantSanitiser, getVariant } from '#lib/utilities/variantHandler';
+import {
+  getPreferredVariant,
+  setPreferredVariant,
+} from '#contexts/UserContext/cookies';
+
+const redirect = (location, service, defaultVariant) => {
+  const preferredVariant = getPreferredVariant(service);
+  return (
+    <Redirect
+      to={{
+        ...location,
+        pathname: `${location.pathname}/${preferredVariant || defaultVariant}`,
+      }}
+    />
+  );
+};
 
-const WithVariant = Component => {
+const withVariant = Component => {
   const VariantContainer = props => {
     const { service, variant } = useParams();
     const location = useLocation();
+    const defaultVariant = getVariant({ service, variant });
+    const sanitizedVariant = variantSanitiser(variant);
+    const isNotDefaultVariant = defaultVariant && defaultVariant !== 'default';
```

Comment:
reiterating from this [comment](https://github.com/bbc/simorgh/pull/4755#discussion_r353626568) 

```suggestion
    const isNotDefaultVariant = defaultVariant !== 'default';
```